src/fontra/views/fontinfo/panel-development-status-definitions.js

```diff
@@ -0,0 +1,325 @@
+    }
+  }
+}
+
+function getStatusFieldDefinitions(fontController) {
+  const statusFieldDefinitions =
+    fontController.customData["fontra.sourceStatusFieldDefinitions"];
+  if (statusFieldDefinitions) {
+    return statusFieldDefinitions;
+  }
+  return [];
+}
+
+addStyleSheet(`
+.fontra-ui-font-info-status-definitions-panel-status-def-box {
+  background-color: var(--ui-element-background-color);
+  border-radius: 0.5em;
+  padding: 1em;
+  cursor: pointer;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: max-content max-content max-content auto;
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-delete {
+  justify-self: end;
+  /*align-self: start;*/
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-color {
+  width: 8em;
+  /*height: 1.6em; without height: best compromise for all browsers*/
+  margin: 0;
+  padding: 0;
+  outline: none;
+  border: none;
+  border-color: transparent;
+  border-radius: 0.25em;
+}
+`);
+
+class StatusDefBox extends HTMLElement {
+  constructor(fontController, statusIndex, postChange, setupUI) {
+    super();
+    this.classList.add("fontra-ui-font-info-status-definitions-panel-status-def-box");
+    this.fontController = fontController;
+    this.statusIndex = statusIndex;
+    this.postChange = postChange;
+    this.setupUI = setupUI;
+    this._updateContents();
+  }
+
+  get statusDef() {
+    return this.fontController.customData["fontra.sourceStatusFieldDefinitions"][
+      this.statusIndex
+    ];
+  }
+
+  replaceStatusDef(newStatusDef, undoLabel, statusIndex = this.statusIndex) {
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"][statusIndex] =
+        newStatusDef;
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  deleteStatusDef(statusIndex) {
+    const undoLabel = `delete status def '${this.statusDef.name}'`;
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"].splice(statusIndex, 1);
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  changeStatusDefIsDefault(statusDef, event) {
+    const undoLabel = `change status definition isDefault`;
+    if (!event.target.checked) {
+      delete statusDef["isDefault"];
+      this.replaceStatusDef(statusDef, undoLabel);
+      return;
+    }
+
+    // If checked: Set all status definitions to false, first.
+    for (const [index, statusDefTemp] of enumerate(
+      this.fontController.customData["fontra.sourceStatusFieldDefinitions"]
+    )) {
+      delete statusDefTemp["isDefault"];
```

Comment:
`statusDefTemp` is real data, not a copy, and you are not supposed to modify that. I think you need something like this here:

```js
    const newStatusDef = {...statusDefTemp};  // Maybe just call it `oldStatusDef` ?
    delete newStatusDef["isDefault"];
```

Reply:
thanks. done.